plugins/modules/ise_radius_integration_workflow_manager.py

```diff
@@ -1163,6 +1164,60 @@ def accept_cisco_ise_server_certificate(self, ipAddress, trusted_server):
             return None
         return
 
+    def check_auth_server_response_status(self, response, validation_string_set):
+        """
+        Format the parameter of the payload for updating the authentication and policy server
+        in accordance with the information in the Cisco Catalyst Ceter.
+
+        Parameters:
+            have_auth_server (dict) - Authentication and policy server information from the Cisco Catalyst Center.
+            want_auth_server (dict) - Authentication and policy server information from the Playbook.
+
+        Returns:
+            self - The current object with updated desired Authentication Policy Server information.
+        """
+
+        response = response.get("response")
+        if response.get("errorcode") is not None:
+            self.msg = response.get("detail")
+            self.status = "failed"
+            return self
+
+        task_id = response.get("taskId")
+        start_time = time.time()
+        while True:
+            end_time = time.time()
+            if (end_time - start_time) >= self.max_timeout:
+                self.msg = "Max timeout of {0} sec has reached for the execution id '{1}'.".format(self.max_timeout, task_id) + \
+                           "Exiting the loop due to unexpected API 'edit_authentication_and_policy_server_access_configuration' status."
+                self.status = "failed"
+                break
+
+            task_details = self.get_task_details(task_id)
+            self.log('Getting task details from task ID {0}: {1}'.format(task_id, task_details), "DEBUG")
+            if task_details.get("isError") is True:
+                failure_reason = task_details.get("failureReason")
+                if failure_reason:
+                    self.msg = str(failure_reason)
+                else:
+                    self.msg = str(task_details.get("progress"))
+                self.status = "failed"
+                break
+
+            for validation_string in validation_string_set:
+                if validation_string in task_details.get("progress").lower():
+                    self.result['changed'] = True
+                    self.validation_string = validation_string
+                    self.status = "success"
```

Comment:
Don't we need to break from this for loop if we found the validation_string in task_details?